### ISSUE TITLE
Consider using GNU General Public License v2.0 instead of MIT & describe the decision process in the documentation

### ISSUE THREAD

**Issue by michael-spengler** [fancy-flashcard/ffc#20]
Perhaps you can enhance and include the following "Ups and Downs" decision table:

MIT vs. GNU GPL in the context of fancy-flashcard
| Option                           | Ups        |  Downs |
| ------------------- | ------------------ | -------------- |
| MIT |  Simplest License    | Potential lack of motivation to publish improvements |
| GNU GPL |  Every party which adds improvements is obliged to publish those improvements. With this might come a quicker positive co-evolution.    | tbd |

**Comment by nikolockenvitz** [comment]
👍 we start working on that soon

**Comment by FischerRene** [comment]
The following idea might be a bit too philosophical:
By using the MIT license we might get more people, especially companies, to use the software and refine it.
Even though we might lose our own benefit by opening it up to everyone like this, the collective gains are potentially higher.

We probably should create our own license that combines the freedom of MIT with a catch for companies that improve it on their own without sharing the results and try to earn money with it.
We see the potential (monetary) success of the app in the size of the user base.
I'd argue that the most creative and innovative developers will always choose to improve an open source product instead of achieving a monetary gain.
This is the same with a friendly and helpful user base &rarr; we might not become the biggest fish in the sea, but we will have loyal users as long as we are loyal to them.

You could compare these ideas to the [ideas behind choosing apache 2.0 as androids license](https://source.android.com/setup/start/licenses#why-apache-software-license).

**Comment by nikolockenvitz** [comment]
@michael-spengler when deciding for MIT License at the beginning of this project we were not so focused on monetary aspects.
Perhaps it's a bit idealistically that we don't want to force others to share the code.

But we get the point that others could build upon our code without disclosing source code and make money with it.
Getting others to share ideas/code/improvements is nice but the actual thing we don't want others to do is earning money with our inventions.
As Rene has implied we think that we can profit from others and their contributions as they help making Fancy Flashcard successful.
To maximize this we want to have as few restrictions as possible.

So in conclusion, it's not about whether others disclose the source code or not but whether they make money with it or not.
In the latter case we would like to participate in this (similarly to what we described in [our thoughts on monetization](https://github.com/fancy-flashcard/ffc/tree/master/docs#monetization)).
What do you think about adjusting our license (still based on MIT) to oblige others to share 5% of their revenues that are based on Fancy Flashcard / our software?

**Comment by nikolockenvitz** [comment]
could be interesting: https://commonsclause.com/

**Comment by michael-spengler** [comment]
Interesting thoughts. 
As my main goal as a lecturer is not saying this one is the best or this one is the best. My main goal is to support a sophisticated and wise discourse which shall increase the likelihood of wise findings and decisions. Therefore I want to add the following perspective to the ones you and I explored above: 

I think especially schools and universities should use **free software** - using non free software in schools and universities can be compared to giving kids cigarettes - injecting a dependency which can be the opposite of healthy.

Even if fancy flashcard itself is free - there could be people with a higher marketing budget injecting this dependency if we do not use the GNU GPL 

Background on this perspective:
https://www.youtube.com/watch?v=Ag1AKIl_2GM

**Comment by nikolockenvitz** [comment]
> 
> 
> Interesting thoughts.
> My main goal as a lecturer is not saying this one is the best or this one is the best. My main goal is to support a sophisticated and wise discourse which shall increase the likelihood of wise findings and decisions. Therefore I want to add the following perspective to the ones you and I explored above:
> 
> I think especially schools and universities should use **free software** - using non free software in schools and universities can be compared to giving kids cigarettes - injecting a dependency which can be the opposite of healthy.
> 
> Even if fancy flashcard itself is free - there could be people with a higher marketing budget injecting this dependency if we do not use the GNU GPL
> 
> Background on this perspective:
> https://www.youtube.com/watch?v=Ag1AKIl_2GM

Interesting ideas. We didn't thought much about enforcing free software before. We think about this in the next days and give you an updaten then.

**Comment by michael-spengler** [comment]
ok cool - as I said - it's up to you of course - I'm just offering additional perspectives without being 100 percent sure about what the perfect answer to this is.

**Comment by FischerRene** [comment]
FYI: We changed the license to GNU GPL v2.0 and documented our thoughts in the documentation right [here](https://github.com/fancy-flashcard/ffc/tree/master/docs#license).
